File: 2021_summer_algorithm_study_intermediate/using_python/16563.py
"""
input :
5
5 4 45 64 54

output :
5
2 2
3 3 5
2 2 2 2 2 2
2 3 3 3
"""

import sys
sys.stdin = open("input.txt", "r")
input = sys.stdin.readline


def factorize(num):
    while True:
        if least_primes[num] == -1:
            print(num, end=' ')
            break
        print(least_primes[num], end=' ')
        num //= least_primes[num]


if __name__ == "__main__":
    N = int(input())
    nums = list(map(int, input().rstrip().split()))

    # 전처리
    # sqrt(5,000,000) < 2240
    primes = []
    least_primes = [-1 for _ in range(5000001)]
    for i in range(2, 2240):
        for j in range(i + i, 5000001, i):
            if least_primes[j] != -1: continue

            least_primes[j] = i
    ##########################################

    for num in nums:
        factorize(num)
        print()

File: 2021_summer_algorithm_study_intermediate/using_cpp/16563.cpp
/*
input :
5
5 4 45 64 54

output :
5
2 2
3 3 5
2 2 2 2 2 2
2 3 3 3
*/

#include <iostream>
#include <algorithm>
using namespace std;

// 전역 변수 선언
int N;
int num;
int least_primes[5000005];


void factorize(int num){
    while (true){
        if (least_primes[num] == 0){
            cout << num << ' ';
            break;
        }
        cout << least_primes[num] << ' ';
        num /= least_primes[num];
    }
}

int main(){
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    freopen("input.txt", "r", stdin);

    cin >> N;
    // 전처리
    for (int i = 2; i < 2240; i++){
        for (int j = i + i; j < 5000001; j += i){
            if (least_primes[j] != 0) continue;
            least_primes[j] = i;
        }
    }
    ///////////////////////////

    while(N--){
        cin >> num;
        factorize(num);
        cout << '\n';
    }

    return 0;
}